onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/split_op_builder.cc

```diff
@@ -0,0 +1,161 @@
+  int GetMinSupportedOpSet(const NodeUnit& /* node_unit */) const override { return 13; }
+
+  // NNAPI Split is available since NNAPI feature level 3
+  int32_t GetMinSupportedNNAPIFeatureLevel(const NodeUnit& /* node_unit */,
+                                           const OpSupportCheckParams& /* params */) const override {
+    return ANEURALNETWORKS_FEATURE_LEVEL_3;
+  }
+};
+
+// Add operator related
+
+void SplitOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
+  const auto& input_defs = node_unit.Inputs();
+
+  if (input_defs.size() > 1 && input_defs[1].node_arg.Exists()) {  // optional second input "split"
+    model_builder.AddInitializerToSkip(input_defs[1].node_arg.Name());
+  }
+}
+
+Status SplitOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
+  const auto& input_name = node_unit.Inputs()[0].node_arg.Name();
+  const auto& outputs = node_unit.Outputs();
+
+  NodeAttrHelper helper(node_unit);
+  const auto axis = helper.Get("axis", 0);
+
+  int32_t num_outputs;
+  if (node_unit.SinceVersion() >= 18) {
+    num_outputs = SafeInt<int32_t>(*helper.GetInt("num_outputs"));
+  } else {
+    num_outputs = SafeInt<int32_t>(node_unit.Outputs().size());
+  }
+
+  std::vector<std::string> output_names;
+  output_names.reserve(num_outputs);
+  for (int32_t i = 0; i < num_outputs; ++i) {
+    output_names.push_back(outputs[i].node_arg.Name());
+  }
+
+  ORT_RETURN_IF_ERROR(op_builder_helpers::AddNnapiSplit(model_builder, input_name, axis, output_names));
+
+  return Status::OK();
+}
+
+// Operator support related
+
+bool SplitOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                                       const OpSupportCheckParams& /* params */) const {
+  Shape input_shape;
+  if (!GetShape(node_unit.Inputs()[0].node_arg, input_shape))
+    return false;
+
+  const auto& input_defs = node_unit.Inputs();
+  NodeAttrHelper helper(node_unit);
+  const auto axis = helper.Get("axis", 0);
+
+  const auto split_dims_at_axis = input_shape[HandleNegativeAxis(axis, input_shape.size())];
+  if (input_defs.size() > 1 && input_defs[1].node_arg.Exists()) {
+    // if optional input `split` is provided
+    auto split_initializer_it = initializers.find(input_defs[1].node_arg.Name());
+    if (split_initializer_it == initializers.end()) {
+      LOGS_DEFAULT(VERBOSE) << "Optional input 'split' must be initializer if provided.";
+      return false;
+    }
+    const auto& splits_tensor = *split_initializer_it->second;
+    Initializer unpacked_tensor(splits_tensor);
+    auto splits_span = unpacked_tensor.DataAsSpan<int64_t>();
+    uint32_t sum_of_splits = std::accumulate(splits_span.begin(), splits_span.end(), SafeInt<uint32_t>(0));
+    if (sum_of_splits != split_dims_at_axis) {
+      LOGS_DEFAULT(VERBOSE) << "Sum of the 'split' input values must equal to the dim value at 'axis' specified.\n"
```

Comment:
nit: I think having newlines in log messages is usually unnecessary. there's useful info prior to the message and it is easier to parse if it's on a single line, IMO.
```suggestion
      LOGS_DEFAULT(VERBOSE) << "Sum of the 'split' input values must equal to the dim value at 'axis' specified. "
```